applications/RomApplication/custom_modelers/hrom_visualization_mesh_modeler.cpp

```diff
@@ -0,0 +1,118 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ruben Zorrilla
+//
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/kratos_parameters.h"
+#include "utilities/parallel_utilities.h"
+#include "utilities/variable_utils.h"
+
+// Application includes
+#include "hrom_visualization_mesh_modeler.h"
+#include "rom_application_variables.h"
+
+namespace Kratos
+{
+
+HRomVisualizationMeshModeler::HRomVisualizationMeshModeler(
+    Model& rModel,
+    Parameters rParameters)
+    : Modeler(rModel, rParameters)
+    , mpHRomModelPart(&(rModel.GetModelPart(rParameters["hrom_model_part_name"].GetString())))
+    , mpVisualizationModelPart(&(rModel.GetModelPart(rParameters["hrom_visualization_model_part_name"].GetString())))
+{
+    // Check default settings
+    this->CheckDefaultSettings(rParameters);
```

Comment:
```suggestion
    rParameters.ValidateAndAssignDefaults(GetDefaultParameters());
```

More Kratos-like.

Reply:
ofc missing here is removing `CheckDefaultSettings` and overriding `GetDefaultParameters`.

Reply:
Done! I also added the `GetDefaultParameters` to the base modeler. See #9716.